File: Tuning/sample_code/03_prof-ex/src/c/message.c
/* Copyright 2024 Research Organization for Information Science and Technology */
/*--------------------------------------------------------------------*/
/* message.c                                                            
   Author: Yukihiro Ota (yota@rist.or.jp)                            
   Last update: 29 July 2017                                          */
/*--------------------------------------------------------------------*/
#include <stdio.h>
#include "setup.h"
#include "message.h"

/*--------------------------------------------------------------------*/
/* message_of_header                                                  */
/*--------------------------------------------------------------------*/
void message_of_header (void)
{
  printf(
    "======================================================\n"
    " diffuse\n" 
    "MOLECULAR DYNAMICS METHOD FOR MOLECULAR DIFFUSION PROBLEM\n"
    "\n"
    "Author: Yukihiro Ota (yota@rist.or.jp)\n"
    "Last update: 28 July 2017\n"
    "\n"   
    "The original code is taken from a textbook,\n"
    "A. Sato, Introduction to Molecular simulation with sample\n"
    "simulation programs (in Japanese) (Kyouritsu, 2004) 3.1.6\n"
    "\n"  
    "======================================================\n");
  return;
}
/*--------------------------------------------------------------------*/
/* message_of_usage                                                   */
/*--------------------------------------------------------------------*/
void message_of_usage ( void )
{
  message_of_header ();

  printf(
    "-------------------------------------------------------\n"
    "[usage] diffuse.x <arg1>\n"
    "  <arg1>: any integer (e.g. 0)\n"
    "          This argument is never used in the program.\n"
    "-------------------------------------------------------\n"
    "<input parameters>\n"
    "num_partot: Total number of particles (NTOT=NA+NB)\n"
    "num_parA: Number of particle A\n"
    "num_parB: Number of particle B\n"
    "nstep: Total number of time steps\n"
    "nobs: Observation step\n"
    "\n"
    "iflag_test: Flag for test mode (0/1)\n"
    "  0: test mode (no real calculation is performed)\n"
    "  1: simulation mode\n"
    "\n"
    "mamu_parA: Mass of particle A (atomic mass unit)\n"
    "mamu_parB: Mass of particle B (atomic mass unit)\n"
    "len_cel: Unit-cell length (square box) (Angstrom)\n"
    "tempt: Temperature (K)\n"
    "epsLJ: epsilon of LJ pot (kJ/mol)\n"
    "sigLJ: sigma of LJ pot (Angstrom)\n"
    "dt: Increment of time step (femto second)\n"
    "\n"); 
  return;
}
/*--------------------------------------------------------------------*/
/* message_of_rephinp                                                 */
/*--------------------------------------------------------------------*/
void message_of_rephinp ( InputData *pt )
{

  puts("[Input parameters]");
  printf("num_partot: Total number of particles (NTOT=NA+NB)\n%10d\n",
		  pt->num_partot);
  printf("num_parA: Number of particle A\n%10d\n",pt->num_parA);
  printf("num_parB: Number of particle B\n%10d\n",pt->num_parB);
  printf("nstep: Total number of time steps\n%10d\n",pt->nstep);
  printf("nobs: Observation step\n%10d\n",pt->nobs);
  printf("\n");
       
  printf("iflag_test: Flag for test mode (0/1)\n%10d\n",
		  pt->iflag_test);
  printf("mamu_parA: Mass of particle A \n%17.6e%18s\n",
		  pt->mamu_parA,"amu");
  printf("mamu_parB: Mass of particle B \n%17.6e%18s\n",
		  pt->mamu_parB,"amu");
  printf("len_cell: Unit-cell length (square box) \n%17.6e%18s\n",
		  pt->len_cell,"Angstrom");
  printf("tempt: Temperature \n%17.6e%18s\n",pt->tempt,"K");
  printf("epsLJ: epsilon of LJ pot \n%17.6e%18s\n",pt->epsLJ,"kJ/mol");
  printf("sigLJ: sigma of LJ pot \n%17.6e%18s\n",pt->sigLJ,"Angstrom");
  printf("dt: Increment of time step \n%17.6e%18s\n",pt->dt,"fs");
  printf("\n");

  return;
}
/*--------------------------------------------------------------------*/
/* message_of_calcparam                                               */
/*--------------------------------------------------------------------*/
void message_of_calcparam ( InputData *pt )
{

  puts("[Calculated parameters]");
  printf("<Unit of length>\n%17.6e%18s\n",pt->lunit, "(Angstrom)");
  printf("<Unit of velocity>\n%17.6e%18s\n",pt->vunit, "(m/sec)");
  printf("<Unit of time>\n%17.6e%18s\n",pt->tunit, "(ns)");
  printf("<Unit of temperature>\n%17.6e%18s\n",pt->temptunit, "(K)");
  printf("<Unit of number density>\n%17.6e%18s\n",
		  pt->ndensunit, "(Angstrom)^-3");
  printf("\n");
       
  printf("<Dimensionless unit-cell length>\n%17.6e\n",pt->leng);
  printf("<Dimensionless number density>\n%17.6e\n",pt->ndens);
  printf("<Dimensionless time increment>\n%17.6e\n",pt->dtu);
  printf("\n");
       
  printf("The following two quantities are useful for\n"
         "tuning <nstep> and <dt>.\n");
  printf("<Characteristic number of time steps>\n"
         "(unit-cell-length/2)/(dt*sqrt(3*temperature))=\n");
  printf("%17.6e\n",pt->nstep_upbd);
  printf("<Characteristic dimensionless time increment>\n"
         "0.1/sqrt(3*temperature)=\n");
  printf("%17.6e\n",pt->dt_upbd);
  printf("\n");
  return;
}
/*--------------------------------------------------------------------*/
/* message_of_output                                                  */
/*--------------------------------------------------------------------*/
void message_of_output ( void ) 
{
  printf(
    "[Explanation on output files]\n"
    "+ inigeom.xyz\n"
    "Initial configuration of particle positions is shown\n"
    "with the XYZ format. Each coordinate is dimensionless.\n"
    "\n"
    "+ trj.xyz\n"
    "Trajectory of particle positions is shown\n"
    "with the XYZ format. Each coordinate is dimensionless\n"
    "The stepsize of the trajectory is controlled by the \n"
    "input parameter, <nobs>.\n"
    "\n");
  return;
}
/*--------------------------------------------------------------------*/
/* message_of_testmode                                                */
/*--------------------------------------------------------------------*/
void message_of_testmode ( InputData *p ) 
{

  message_of_header ();

  message_of_rephinp ( p );

  printf("--TEST-MODE-TEST-MODE-TEST-MODE-TEST-MODE-TEST-MODE--\n");

  message_of_calcparam ( p ); 

  message_of_output ();

  return;
}
/*--------------------------------------------------------------------*/
/* message_of_abtermination                                           */
/*--------------------------------------------------------------------*/
void message_of_abtermination ( void )
{
   printf("*** The simulation is abnormally terminated ***\n");
   return;
}

File: Tuning/sample_code/03_prof-ex/src/fortran/message.f90
! Copyright 2024 Research Organization for Information Science and Technology
!!$---------------------------------------------------------------------
!!$ message.f90
!!$ Author: Yukihiro Ota (yota@rist.or.jp)
!!$ Last update: 1 Nov 2016
!!$
!!$---------------------------------------------------------------------

!!$---Subroutine: message_of_header
subroutine message_of_header

  use common_par,only:unit_std_out
  
  implicit none
  
  write(unit_std_out,'(A)') &
       "======================================================="
  write(unit_std_out,'(A)') &
       " diffuse"
  write(unit_std_out,'(A)') &
       "MOLECULAR DYNAMICS METHOD FOR MOLECULAR DIFFUSION PROBLEM"
  write(unit_std_out,'(A)') 
       
  write(unit_std_out,'(A)') &
       "Author: Yukihiro Ota (yota@rist.or.jp)"
  write(unit_std_out,'(A)') &
       "Last update: 1 Nov 2016"
  write(unit_std_out,'(A)') 
       
  write(unit_std_out,'(A)') &
       "The original code is taken from a textbook,"
  write(unit_std_out,'(A)') &
       "A. Sato, Introduction to Molecular simulation with sample"
  write(unit_std_out,'(A)') &
       "simulation programs (in Japanese) (Kyouritsu, 2004) 3.1.6"
  write(unit_std_out,'(A)') 
       
  write(unit_std_out,'(A)') &
       "======================================================="
  write(unit_std_out,'(A)') 

  return
end subroutine message_of_header
!!$---End of Subroutine: message_of_header


!!$---Subroutine: message_of_usage
subroutine message_of_usage

  use common_par,only:unit_std_out
  
  implicit none
 
  call message_of_header

  write(unit_std_out,'(A)') &
       "-------------------------------------------------------"
  write(unit_std_out,'(A)') &
       "[usage] diffuse.x <arg1> "
  write(unit_std_out,'(A)') &
       "  <arg1>: any integer (e.g. 0)"
  write(unit_std_out,'(A)') &
       "          This argument is never used in the program.  "
  write(unit_std_out,'(A)') &
       "-------------------------------------------------------"
  
  write(unit_std_out,'(A)') &
       "<input parameters>"
  write(unit_std_out,'(A)') &
       "num_partot: Total number of particles (NTOT=NA+NB)"
  write(unit_std_out,'(A)') &
       "num_parA: Number of particle A"
  write(unit_std_out,'(A)') &
       "num_parB: Number of particle B"
  write(unit_std_out,'(A)') &
       "nstep: Total number of time steps"
  write(unit_std_out,'(A)') &
       "nobs: Observation step"
  write(unit_std_out,'(A)') 
       
  write(unit_std_out,'(A)') &
       "iflag_test: Flag for test mode (0/1)"
  write(unit_std_out,'(A)') &
       "  0: test mode (no real calculation is performed)"
  write(unit_std_out,'(A)') &
       "  1: simulation mode"
  write(unit_std_out,'(A)') 
       
  write(unit_std_out,'(A)') &
       "mamu_parA: Mass of particle A (atomic mass unit)"
  write(unit_std_out,'(A)') &
       "mamu_parB: Mass of particle B (atomic mass unit)"
  write(unit_std_out,'(A)') &
       "len_cel: Unit-cell length (square box) (Angstrom)"
  write(unit_std_out,'(A)') &
       "tempt: Temperature (K)"
  write(unit_std_out,'(A)') &
       "epsLJ: epsilon of LJ pot (kJ/mol)"
  write(unit_std_out,'(A)') &
       "sigLJ: sigma of LJ pot (Angstrom)"
  write(unit_std_out,'(A)') &
       "dt: Increment of time step (femto second)"
  write(unit_std_out,'(A)') 
       
  return
end subroutine message_of_usage
!!$---End of Subroutine: message_of_usage


!!$---Subroutine: message_of_rephinp
subroutine message_of_rephinp

  use common_par,only: num_partot,num_parA,num_parB,nstep,nobs,&
       iflag_test,mamu_parA,mamu_parB,len_cell,tempt,epsLJ,    &
       sigLJ,dt,unit_std_out
  
  implicit none
  
  write(unit_std_out,'(A)') &
       "[Input parameters]"
  write(unit_std_out,'(A)') &
       "num_partot: Total number of particles (NTOT=NA+NB)"
  write(unit_std_out,'(1I10)') &
       num_partot
  write(unit_std_out,'(A)') &
       "num_parA: Number of particle A"
  write(unit_std_out,'(1I10)') &
       num_parA
  write(unit_std_out,'(A)') &
       "num_parB: Number of particle B"
  write(unit_std_out,'(1I10)') &
       num_parB
  write(unit_std_out,'(A)') &
       "nstep: Total number of time steps"
  write(unit_std_out,'(1I10)') &
       nstep
  write(unit_std_out,'(A)') &
       "nobs: Observation step"
  write(unit_std_out,'(1I10)') &
       nobs
  write(unit_std_out,'(A)') 
       
  write(unit_std_out,'(A)') &
       "iflag_test: Flag for test mode (0/1)"
  write(unit_std_out,'(1I10)') &
       iflag_test
  write(unit_std_out,'(A)') &
       "mamu_parA: Mass of particle A "
  write(unit_std_out,'(1E17.6,1A18)') &
       mamu_parA,"amu"
  write(unit_std_out,'(A)') &
       "mamu_parB: Mass of particle B "
  write(unit_std_out,'(1E17.6,1A18)') &
       mamu_parB,"amu"
  write(unit_std_out,'(A)') &
       "len_cell: Unit-cell length (square box) "
  write(unit_std_out,'(1E17.6,1A18)') &
       len_cell,"Angstrom"
  write(unit_std_out,'(A)') &
       "tempt: Temperature "
  write(unit_std_out,'(1E17.6,1A18)') &
       tempt,"K"
  write(unit_std_out,'(A)') &
       "epsLJ: epsilon of LJ pot "
  write(unit_std_out,'(1E17.6,1A18)') &
       epsLJ,"kJ/mol"
  write(unit_std_out,'(A)') &
       "sigLJ: sigma of LJ pot "
  write(unit_std_out,'(1E17.6,1A18)') &
       sigLJ,"Angstrom"
  write(unit_std_out,'(A)') &
       "dt: Increment of time step "
  write(unit_std_out,'(1E17.6,1A18)') &
       dt,"fs"
  write(unit_std_out,'(A)') 
       
  return
end subroutine message_of_rephinp
!!$---End of Subroutine: message_of_rephinp


!!$---Subroutine: message_of_calcparam
subroutine message_of_calcparam

  use common_par,only: unit_std_out, &
       lunit,vunit,tunit,temptunit,ndensunit,&
       leng,ndens,dtu,nstep_upbd,dt_upbd
 
  implicit none

  write(unit_std_out,'(1A)') &
       "[Calculated parameters]"
  write(unit_std_out,'(1A)') &
       "<Unit of length>"
  write(unit_std_out,'(1E17.6,1A18)') &
       lunit, "(Angstrom)"
  write(unit_std_out,'(1A)') &
       "<Unit of velocity>"
  write(unit_std_out,'(1E17.6,1A18)') &
       vunit, "(m/sec)"
  write(unit_std_out,'(1A)') &
       "<Unit of time>"
  write(unit_std_out,'(1E17.6,1A18)') &
       tunit, "(ns)"
  write(unit_std_out,'(1A)') &
       "<Unit of temperature>"
  write(unit_std_out,'(1E17.6,1A18)') &
       temptunit, "(K)"
  write(unit_std_out,'(1A)') &
       "<Unit of number density>"
  write(unit_std_out,'(1E17.6,1A18)') &
       ndensunit, "(Angstrom)^-3"
  write(unit_std_out,'(A)') 
       
  write(unit_std_out,'(1A)') &
       "<Dimensionless unit-cell length>"
  write(unit_std_out,'(1E17.6)') &
       leng
  write(unit_std_out,'(1A)') &
       "<Dimensionless number density>"
  write(unit_std_out,'(1E17.6)') &
       ndens
  write(unit_std_out,'(1A)') &
       "<Dimensionless time increment>"
  write(unit_std_out,'(1E17.6)') &
       dtu
  write(unit_std_out,'(1A)') 
       
  write(unit_std_out,'(1A)') &
       "The following two quantities are useful for"
  write(unit_std_out,'(1A)') &
       "tuning <nstep> and <dt>."
  write(unit_std_out,'(1A)') &
       "<Characteristic number of time steps>"
  write(unit_std_out,'(1A)') &
       "(unit-cell-length/2)/(dt*sqrt(3*temperature))="
  write(unit_std_out,'(1E17.6)') &
       nstep_upbd
  write(unit_std_out,'(1A)') &
       "<Characteristic dimensionless time increment>"
  write(unit_std_out,'(1A)') &
       "0.1/sqrt(3*temperature)="
  write(unit_std_out,'(1E17.6)') &
       dt_upbd
  write(unit_std_out,'(A)') 

  return
end subroutine message_of_calcparam
!!$---End of Subroutine: message_of_calcparam


!!$---Subroutine: message_of_output
subroutine message_of_output

  use common_par,only: unit_std_out
  
  implicit none

  write(unit_std_out,'(A)') &
       "[Explanation on output files]"
  write(unit_std_out,'(A)') &
       "+ inigeom.xyz"
  write(unit_std_out,'(A)') &
       "Initial configuration of particle positions is shown"
  write(unit_std_out,'(A)') &
       "with the XYZ format. Each coordinate is dimensionless."
  write(unit_std_out,'(A)') 
       
  write(unit_std_out,'(A)') &
       "+ trj.xyz"
  write(unit_std_out,'(A)') &
       "Trajectory of particle positions is shown"
  write(unit_std_out,'(A)') &
       "with the XYZ format. Each coordinate is dimensionless"
  write(unit_std_out,'(A)') &
       "The stepsize of the trajectory is controlled by the "
  write(unit_std_out,'(A)') &
       "input parameter, <nobs>."
  write(unit_std_out,'(A)') 
       

  return
end subroutine message_of_output
!!$---End of Subroutine: message_of_output


!!$---Subroutine: message_of_testmode
subroutine message_of_testmode

  use common_par,only: unit_std_out
    
  implicit none
  
  call message_of_header

  call message_of_rephinp

  write(unit_std_out,'(A)') &
       "--TEST-MODE-TEST-MODE-TEST-MODE-TEST-MODE-TEST-MODE--"

  call message_of_calcparam

  call message_of_output
  
  return
end subroutine message_of_testmode
!!$---End of Subroutine: message_of_testmode


!!$=--Subroutine: message_of_abtermination
subroutine message_of_abtermination

  use common_par,only: unit_std_out
  
  implicit none

  write(unit_std_out,'(A)') &
       "*** The simulation is abnormally terminated ***"
  return
end subroutine message_of_abtermination
!!$=--End of Subroutine: message_of_abtermination

